### PR TITLE
Fixed Simulation.exports setter

### LINKED ISSUE #833: [BUG] `Simulation.exports` should always be a list
**Describe the bug**
FESTIM expects a list for the exports attribute of `Simulation`.

However, no error is raised when users give something else.

**To Reproduce**

Run on main:

```python
import numpy as np
import festim as F

my_model = F.Simulation()
my_model.mesh = F.MeshFromVertices(vertices=np.linspace(0, 1, num=100))
my_model.materials = F.Material(id=1, D_0=1, E_D=0, Q=2)
my_model.T = F.Temperature(value=700 + 1000 * F.x)
my_model.settings = F.Settings(
    absolute_tolerance=1e-10, relative_tolerance=1e-10, transient=False, soret=True
)

flux_left = F.SurfaceFlux(field=0, surface=1)
flux_right = F.SurfaceFlux(field=0, surface=2)
my_model.exports = F.DerivedQuantities([flux_left, flux_right])

my_model.initialise()
my_model.run()

print(flux_left.data)
print(flux_right.data)
```

Produces an empty list for data of derived quantities (they are just ignored).

```
Defining initial values
Defining variational problem
Defining source terms
Defining boundary conditions
Solving steady state problem...
Solved problem in 0.00 s
[]
[]
```

**Expected behavior**
Either the exports setter should convert the user input to a list, or an error should be raised

### PR DESCRIPTION
**Pull request by RemDelaporteMathurin**
## Proposed changes

Fixes #833 

## Types of changes

What types of changes does your code introduce to FESTIM?

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [ ] New tests

## Checklist

- [x] Black formatted
- [x] Unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by RemDelaporteMathurin** [comment]
@KulaginVladimir would you mind giving this a quick look? I think this is a result of these classes now inheriting from `list`

File: festim/generic_simulation.py [hunk 1]
```diff
@@ -154,12 +154,12 @@ def exports(self):
...
+            self._exports = festim.Exports([value])
         elif isinstance(value, festim.Exports):
             self._exports = value
         elif isinstance(value, list):
             self._exports = festim.Exports(value)
-        elif isinstance(value, festim.Export):
-            self._exports = festim.Exports([value])
         else:
             raise TypeError(
                 "accepted types for exports are list, festim.Export or festim.Exports"
```
Comment (KulaginVladimir) [inline festim/generic_simulation.py:165]: This line should be changed since derived quantities could be passed, no?

**Review by KulaginVladimir — COMMENTED** [review]
The fix should work 👌🏻

Unfortunately, I can't check it thoroughly at the moment. As I understand, the problem was only with derived quantities, right?

File: festim/generic_simulation.py (hunk 1 above)
Comment (RemDelaporteMathurin) [reply festim/generic_simulation.py:165]: Yep, i just bumped it up a few lines and changed the type check

Comment (KulaginVladimir) [reply festim/generic_simulation.py:165]: I meant the error message. With your fix we can pass `festim.DerivedQunatities` directly to `Simulation.exports`, right?

Comment (RemDelaporteMathurin) [reply festim/generic_simulation.py:165]: Yes! Good shout!

**Review by KulaginVladimir — APPROVED** [review]

**Pull request merged**